src/lib.rs

```diff
@@ -523,6 +524,22 @@ impl<'a, R: Read> Iterator for FilesMut<'a, R> {
     }
 }
 
+impl Clone for Header {
+    fn clone(&self) -> Header {
+        let mut header = Header::new();
+        unsafe { ptr::copy_nonoverlapping(self, &mut header as *mut Header, 1) };
+        header
+    }
+}
+
+impl PartialEq for Header {
+    fn eq(&self, other: &Header) -> bool {
+        self.as_bytes()[..] == other.as_bytes()[..]
```

Comment:
Hmm, what would you suggest happens if a field access fails with error?
You raise an interesting point, because field comparison is not what I'd want but I can see others would.

Maybe `compare_{field,bytes}` methods which people can implement Eq on downstream if they want?